earth_enterprise/src/portableserver/build.py

```diff
@@ -0,0 +1,252 @@
+            self.source_dir, 'fusion', 'portableglobe', 'servers')
+
+        exclude_entries = ['linux', 'mac', 'windows', 'fileunpacker']
+        entries = [f
+                   for f in os.listdir(servers_dir)
+                   if f.lower() not in exclude_entries]
+        copy_from_dir_to_dir(servers_dir, self.server_dir, entries=entries)
+        copy_from_dir_to_dir(
+            os.path.join(servers_dir, self.platform), self.server_dir)
+
+        # Copy from <maps/>:
+        copy_from_dir_to_dir(
+            os.path.join(self.source_dir, 'maps'),
+            os.path.join(self.server_dir, 'local', 'maps'),
+            exclude_entries=['SConscript'])
+
+        self.create_version_txt()
+
+        # Copy configuration files:
+        copy_from_dir_to_dir(self.resources_dir, self.server_dir)
+
+    def create_version_txt(self):
+        """Creates <server/local/version.txt>."""
+
+        self.get_version()
+        path = os.path.join(self.server_dir, 'local', 'version.txt')
+        with open(path, 'w') as output_file:
+            output_file.write(self.version_string)
+
+    def get_version(self):
+        """Parses version information, and sets member variables."""
+
+        if self.base_version is None:
+            path = os.path.join(self.source_dir, 'fusion_version.txt')
+            with open(path, 'r') as input_file:
+                while True:
+                    base_version = input_file.readline().strip()
+                    if not base_version.startswith('#'):
+                        break
+            self.base_version = base_version
+
+        if self.build_date is None:
+            self.build_date = datetime.date.today()
+
+        if self.version_string is None:
+            self.version_string = '{}-{}'.format(
+                self.base_version, self.build_date.strftime('%Y%m%d'))
+
+    def build_fileunpacker(self):
+        """Builds `fileunpacker` shared libraries, and copies them to
+        <server/>."""
+
+        # Keep the source directory clean:
+        task_build_dir = os.path.join(self.build_dir, 'fileunpacker-build')
+        shutil.rmtree(task_build_dir, ignore_errors=True)
+        distutils.dir_util.copy_tree(
+            os.path.join(self.source_dir, 'fusion', 'portableglobe', 'servers',
+                         'fileunpacker'),
+            task_build_dir)
+
+        # Execute 'build_and_test.py':
+        os.chdir(task_build_dir)
+        old_path = sys.path
+        sys.path = [task_build_dir] + sys.path
+        import build_and_test
+        sys.path = old_path
+
+        build_and_test.main(['build_and_test.py', self.platform])
+
+        # Copy library to package directory:
+        exclude_entries = ['test.py', 'util.py']
+        dist_dir = os.path.join(task_build_dir, 'dist')
+        entries = [
+            f
+            for f in os.listdir(dist_dir)
+            if os.path.splitext(f)[1].lower() in
+            ['.so', '.pyd', '.dll', '.py']
```

Comment:
I don't see real benefit from including <.pyc>, and <.pyo>. They'll be generated the first time someone executes the Python scripts anyway.  I'd rather keep the archive clean.